### PR TITLE
acknowledge --show-all=false with --watch

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Allow the `--show-all=false` option to work when invoking `kubectl get ...` with the `--watch` option enabled.

Related downstream issue: https://github.com/openshift/origin/issues/14975

cc @fabianofranz @kubernetes/sig-cli-misc @kargakis

### CODE REVIEW

File: pkg/kubectl/cmd/get.go [hunk 1]
```diff
@@ -278,8 +278,12 @@ func RunGet(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args [
 				objsToPrint = append(objsToPrint, obj)
 			}
 			for _, objToPrint := range objsToPrint {
-				if err := printer.PrintObj(objToPrint, writer); err != nil {
-					return fmt.Errorf("unable to output the provided object: %v", err)
+				if isFiltered, err := filterFuncs.Filter(objToPrint, filterOpts); !isFiltered {
+					if err != nil {
+						glog.V(2).Infof("Unable to filter resource: %v", err)
```
Comment (ericchiang) [inline pkg/kubectl/cmd/get.go]: Why not just return an error here or use glog.Errorf?

**Comment by kargakis** [comment]
@kubernetes/sig-cli-pr-reviews

File: pkg/kubectl/cmd/get.go (hunk 1 above)
Comment (juanvallejo) [reply pkg/kubectl/cmd/get.go]: Took this from the existing filter [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/get.go#L407), but can switch to `glog.Errorf` if that works better. Returning an error will terminate early, potentially not printing any remaining resources (or a filter error for each of them)

Comment (ericchiang) [reply pkg/kubectl/cmd/get.go]: As long as there's precedence, that works for me.

**Comment by juanvallejo** [comment]
cc @fabianofranz

File: pkg/kubectl/cmd/get.go (hunk 1 above)
Comment (fabianofranz) [reply pkg/kubectl/cmd/get.go]: We don't use `glog` in the CLI for anything other that leveled logging (`glog.V(n)...`) .

**Pull request merged**